[gohlucas] iP

gohlucas opened the pull request:
# Dexter
> “Preparation is vital. No detail can be overlooked `[REDACTED]`” – Dexter Morgan ([source](https://www.imdb.com/title/tt0757186/quotes/?item=qt0315167))

Dexter makes the most of your time by keeping track of the things you need to do and ensure that you are prepared. It's:

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/gohlucas/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 🔥
And it is **FREE!**

Features:

 - [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon) 

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

gohlucas committed:
lvl 0

gohlucas committed:
lvl 0

gohlucas committed:
lvl1

gohlucas committed:
lvl1

gohlucas committed:
lvl2

gohlucas committed:
lvl2

gohlucas committed:
lvl3

gohlucas committed:
lvl3

gohlucas committed:
lvl 4 implementation

gohlucas committed:
lvl 4 implementation

gohlucas committed:
TextUiTesting

gohlucas committed:
TextUiTesting

gohlucas committed:
in progress error catch

gohlucas committed:
Merge branch 'my-work'

gohlucas committed:
functionality for level-5

gohlucas committed:
Level 6 del

gohlucas committed:
backup code

gohlucas committed:
branch-Level-7

gohlucas committed:
Merge branch 'branch-Level-7'

gohlucas committed:
lvl 8 stuff

gohlucas committed:
Merge branch 'branch-Level-8'

gohlucas committed:
reshuffled order

gohlucas committed:
Merge branch 'A-MoreOOP'

gohlucas committed:
oop stuff

gohlucas committed:
packages stuff

gohlucas committed:
aa

gohlucas committed:
Merge branch 'A-Packages'

gohlucas committed:
Merge branch 'add-gradle-support'

gohlucas committed:
lvl 7 fix

gohlucas committed:
Merge branch 'branch-Level-7-fix'

gohlucas committed:
saving progress

gohlucas committed:
Merge branch 'A-Gradle'

gohlucas committed:
Add build.gradle to commits

gohlucas committed:
Add JUnit tests

gohlucas committed:
temp

gohlucas committed:
temp1

gohlucas committed:
Fix bug for JUnit commit

gohlucas committed:
Merge branch 'A-JUnit'

gohlucas committed:
Update build gradle details

gohlucas committed:
Implement JavaDocs

gohlucas committed:
Fit to coding standards

gohlucas committed:
Add Find function

gohlucas committed:
Merge branch 'refs/heads/A-CodingStandard'

gohlucas committed:
Resolve merge conflict between A-JavaDoc and A-CodingStandard

gohlucas committed:
Resolved merge conflict due to branch-Level-9

isaacchua0309 commented on src/main/java/dexter/ui/Ui.java:
```diff
@@ -150,13 +150,19 @@ public TaskList run(TaskList tasks) {
                         if (pp == null) {
                             continue;
```
// ⚠️ Avoid unnecessary `continue` if no further processing occurs.

isaacchua0309 commented on src/main/java/dexter/ui/Ui.java:
```diff
@@ -150,13 +150,19 @@ public TaskList run(TaskList tasks) {
+                        if (z.isDue(pp)) {System.out.println("\t" + z);
                         }
                     }
                     System.out.println(LINE);
```
// ⚠️ LINE should be a well-named constant (e.g., `SEPARATOR_LINE` for clarity).

isaacchua0309 commented on src/main/java/dexter/ui/Ui.java:
```diff
@@ -150,13 +150,19 @@ public TaskList run(TaskList tasks) {
+                     TaskList e = tasks.findKeyword(descript);
+                    System.out.println("\t Here are the matching tasks in your list:");
+                    System.out.println(e);
+                    continue;
```
// ⚠️ Redundant `continue`; consider restructuring.

isaacchua0309 commented on src/main/java/dexter/ui/Ui.java:
```diff
@@ -150,13 +150,19 @@ public TaskList run(TaskList tasks) {
+                }
+
+                else {
                     handleExcept(" ");
```
// ⚠️ Avoid passing `" "` (empty space) if it's not meaningful.

isaacchua0309 commented on src/main/java/dexter/ui/Ui.java:
```diff
@@ -168,7 +174,6 @@ public TaskList run(TaskList tasks) {
                     }
                     String reply = LINE + ans + "\t" + t.toString();
                     int siz = tasks.size();
```
// ⚠️ `siz` is an unclear variable name; rename to `taskCount` for better readability.

isaacchua0309 reviewed (commented):
isaacchua03009 PR review 2 on gohlucas

gohlucas committed:
Fix styling using checkstyle

gohlucas committed:
Merge branch 'A-CheckStyle'

gohlucas committed:
Update checkstyle

gohlucas committed:
Port files for GUI

gohlucas committed:
Add GUI usage

gohlucas committed:
Merge branch 'branch-Level-10'

gohlucas committed:
Rectify small change and align with master before handling future branch creation

gohlucas committed:
Refactor code to align with Coding Quality Standards

There are several repeated methods which can be made more concise by extracting them into methods

Creating a method to handle these instances makes the code neater by adhering to SLAP and prevent DRY

Refactoring the code makes the code neater, easier to understand and makes subsequent changes much easier

gohlucas committed:
Add assertions to mitigate methods

This is to mitigate any potential logic errors on programmer's end

gohlucas committed:
Merge pull request #1 from gohlucas/branch-A-CodeQuality

Refactor code to align with Coding Quality Standards

gohlucas committed:
Merge branch 'master' into branch-A-Assertions

gohlucas committed:
Merge pull request #2 from gohlucas/branch-A-Assertions

Add assertions to mitigate methods

gohlucas committed:
Add extension of D-Places

gohlucas committed:
Merge branch 'branch-BCD-Extension'

gohlucas committed:
Add more error handling methods and help function

gohlucas committed:
Merge branch 'A-MoreErrorHandling'

gohlucas committed:
Update README.md and added Ui.png

gohlucas committed:
Update Ui.png

gohlucas committed:
Update Ui.png image as text was unreadble
